File: cpp/09proc/Proc12.cpp
#include <iostream>
using namespace std;

void SortInc3(double& A, double& B, double& C);
void Minmax(double& a, double& b);
void Swap(double& x, double& y);

int main()
{
	//Task("Proc12");
	double A, B, C;

	for (int i = 0; i < 2; i++)
	{
		cin >> A >> B >> C;
		SortInc3(A, B, C);
		cout << A << B << C;
	}
	
	return 0;
}

void SortInc3(double& A, double& B, double& C)
{
	Minmax(A, B);
	Minmax(A, C);
	Minmax(B, C);
}

void Minmax(double& a, double& b)
{
	if (a > b)
		Swap(a, b);
}

void Swap(double& x, double& y)
{
	x += y;
	y = x - y;
	x -= y;
}

File: php/09proc/proc12.php
<?php
function Swap(&$A, &$B){
        if($A > $B){
                $C = $A;
                $A = $B;
                $B = $C;
        }
}
function SortInc3(&$A, &$B, &$C){
        Swap($A, $B);
        Swap($B, $C);
        Swap($A, $B);
}
$hisob = 1;
while($hisob <= 2){
	$A = mt_rand(-100, 100);
	$B = mt_rand(-100, 100);
	$C = mt_rand(-100, 100);
	echo "A".$hisob." = ".$A;
	echo "<br>B".$hisob." = ".$B;
	echo "<br>C".$hisob." = ".$C;
	SortInc3($A, $B, $C);
	echo "<br>Пас аз иҷрои амалиёт:";
	echo "<br>A".$hisob." = ".$A;
	echo "<br>B".$hisob." = ".$B;
	echo "<br>C".$hisob." = ".$C."<br><br>";
	$hisob++;
}
?>